### PR TITLE
issue/2647 Converted AdaptModel to ES Class

### PR DESCRIPTION
**Pull request by oliverfoster**
#2647 and #2648
* conversion of core AdaptModel to an ES6 Class

### CODE REVIEW

File: src/core/js/models/adaptModel.js [hunk 1]
```diff
@@ -4,45 +4,51 @@ define([
...
+    }
+
+    bubblingEvents() {
+      return [
+        'change:_isComplete',
+        'change:_isInteractionComplete'
+      ];
+    }
+
+    initialize () {
```
Comment (brian-learningpool) [inline src/core/js/models/adaptModel.js]: Minor: space after function name.

File: src/core/js/models/adaptModel.js [hunk 2]
```diff
@@ -555,38 +558,32 @@ define([
...
+      const siblingsCollection = new Backbone.Collection(siblings);
       this.set('_siblings', siblingsCollection);
       return siblingsCollection;
-    },
-
-    setOnChildren: function (key, value, options) {
+    }
 
-      var args = arguments;
+    setOnChildren(key, value, options) {
```
Comment (tomgreenfield) [inline src/core/js/models/adaptModel.js]: Suggest `setOnChildren(...args)` or similar? Would avoid the linter complaining about variables that are never used.

File: src/core/js/models/adaptModel.js [hunk 3]
```diff
@@ -555,38 +558,32 @@ define([
...
-    },
+    }
 
     /**
      * @deprecated since v3.2.3 - please use `model.set('_isOptional', value)` instead
      */
-    setOptional: function(value) {
-      Adapt.log.warn("DEPRECATED - Use model.set('_isOptional', value) as setOptional() may be removed in the future");
+    setOptional(value) {
+      Adapt.log.warn('DEPRECATED - Use model.set(\'_isOptional\', value) as setOptional() may be removed in the future');
```
Comment (tomgreenfield) [inline src/core/js/models/adaptModel.js]: Suggest template literal to irradicate need for escaping:
```suggestion
      Adapt.log.warn(`DEPRECATED - Use model.set('_isOptional', value) as setOptional() may be removed in the future`);
```

File: src/core/js/models/adaptModel.js (hunk 2 above)
Comment (oliverfoster) [reply src/core/js/models/adaptModel.js]: Hmmm...

**Pull request merged**